### ISSUE TITLE
#[derive(PrettyDebug)]

### ISSUE THREAD

**Issue by remexre** [remexre/sparkly-rs#1]
Once the core crate is finished, it'd be nice to have a derive for a `{:#?}`-like Debug, possibly with type-based coloring? So something "stringish" would be green, "intish" would be yellow, "boolish" would be cyan, etc.

It should also be infallible, so printing something of type `struct Foo { a: usize, b: &'static str, c: DoesNotImplDebug, d: bool }` should print:

```
Foo {
    a: 0,
    b: "hello",
    d: false,
    // and one other field
}
```

(for the "one" in "and one other field", https://crates.io/crates/english-numbers ?)

**Comment by remexre** [comment]
Also, it might be possible to do a default impl for all Serialize types? Pending https://github.com/rust-lang/rust/issues/31844 and https://github.com/rust-lang/rust/issues/37653, of course.
